### PR TITLE
add geojson endpoint for observation plan

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds observation plan plotting capabilities.

### CODE REVIEW

**Comment by mcoughlin** [comment]
@guynir42 your suggestion won the day.

File: skyportal/handlers/api/observation_plan.py [hunk 1]
```diff
@@ -820,3 +820,57 @@ def delete(self, observation_plan_request_id):
...
+            .joinedload(EventObservationPlan.planned_observations)
+            .joinedload(PlannedObservation.field)
+            .undefer(InstrumentField.contour_summary)
+        ]
+
+        observation_plan_request = ObservationPlanRequest.get_if_accessible_by(
+            observation_plan_request_id,
+            self.current_user,
+            mode="read",
+            raise_if_none=True,
```
Comment (guynir42) [inline skyportal/handlers/api/observation_plan.py:857]: Convert to check if no results then `self.error`

File: static/js/components/ObservationPlanRequestLists.jsx [hunk 2]
```diff
@@ -78,11 +81,89 @@ const getMuiTheme = (theme) =>
...
+
+  const [obsList, setObsList] = useState(null);
+  useEffect(() => {
+    const fetchObsList = async () => {
+      const response = await dispatch(
+        GET(
+          `/api/observation_plan/${observationplanRequest.id}/geojson`,
+          "skyportal/FETCH_OBSERVATION_PLAN_GEOJSON"
+        )
+      );
```
Comment (guynir42) [inline static/js/components/ObservationPlanRequestLists.jsx:108]: Whats the advantage of using this instead of a fetch defined in the js file?

**Review by guynir42 — COMMENTED** [review]
Some comments below:

File: static/js/components/ObservationPlanRequestLists.jsx (hunk 2 above)
Comment (mcoughlin) [reply static/js/components/ObservationPlanRequestLists.jsx:108]: That assigns a state variable no?

File: skyportal/handlers/api/observation_plan.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/observation_plan.py]: good point

File: static/js/components/ObservationPlanRequestLists.jsx (hunk 2 above)
Comment (guynir42) [reply static/js/components/ObservationPlanRequestLists.jsx:108]: Yes I think you're right.

**Comment by guynir42** [comment]
@mcoughlin I have some doubts about how we've been dealing with `raise_if_none` recently. But that's maybe an issue to handle separately.

**Review by guynir42 — APPROVED** [review]

**Comment by mcoughlin** [comment]
@guynir42 Yeah maybe at the same time with this permissions stuff.

**Pull request merged**